dashboard/assets/packages/ui/src/components/Chart/Chart.tsx

```diff
@@ -2,84 +2,42 @@
 //
 // SPDX-License-Identifier: AGPL-3.0-only
 
-import React, { useRef, useState, useLayoutEffect } from "react"
+import React, { Fragment } from "react"
+import { useElementSize } from "usehooks-ts"
 import "uplot/dist/uPlot.min.css"
-import uPlot, { type AlignedData, type Options, type Series } from "uplot"
+import uPlot, { type AlignedData } from "uplot"
 import UplotReact from "uplot-react"
-import { tooltipPlugin, format, dateFormats, Panel, SeriesPlot } from "@xk6-dashboard/view"
+import { Panel, SeriesPlot } from "@xk6-dashboard/view"
 
-import { common, grey } from "theme/colors.css"
 import { createColorScheme } from "utils"
 import { useDigest } from "store/digest"
 import { useTheme } from "store/theme"
+import { Flex } from "components/Flex"
 import { Grid } from "components/Grid"
+import { Icon } from "components/Icon"
+import { Paper } from "components/Paper"
+import { Tooltip } from "components/Tooltip"
 
+import { createOptions } from "./Chart.utils"
 import * as styles from "./Chart.css"
 
-const sync = uPlot.sync("chart")
-
 interface ChartProps {
   panel: Panel
+  container?: boolean
 }
 
-export default function Chart({ panel }: ChartProps) {
-  const [width, setWidth] = useState(0)
-  const ref = useRef<HTMLDivElement>(null)
+export default function Chart({ panel, container }: ChartProps) {
   const digest = useDigest()
   const { theme } = useTheme()
-
-  const tooltipColor = theme == "dark" ? common.black : common.white
-  const gridColor = theme == "dark" ? grey[900] : grey[100]
-  const axesColor = theme == "dark" ? common.white : common.black
-
-  useLayoutEffect(() => {
-    const updateWidth = () => {
-      if (ref.current) {
-        setWidth(ref.current.offsetWidth)
-      }
-    }
-
-    updateWidth()
-    window.addEventListener("resize", updateWidth)
-
-    return () => window.removeEventListener("resize", updateWidth)
-  })
+  const [ref, { width }] = useElementSize()
 
   const plot = new SeriesPlot(digest, panel, createColorScheme(theme))
 
   if (plot.empty) {
-    return <div ref={ref} />
+    return null
   }
 
-  const options: Options = {
-    class: styles.uplot,
-    width: width,
-    height: 250,
-    title: panel.title,
-    cursor: { sync: { key: sync.key } },
-    legend: { live: false },
-    series: plot.series as Series[],
-    axes: [{}],
-    plugins: [tooltipPlugin(tooltipColor)]
-  }
-
-  options.axes = plot.samples.units.map((unit) => {
-    return {
-      stroke: axesColor,
-      grid: { stroke: gridColor, width: 1 },
-      ticks: { stroke: gridColor },
-      values: (_, ticks) => ticks.map((val) => format(unit, val)),
-      size: 70,
-      scale: unit
-    }
-  })
-
-  delete options.axes[0].size
-  options.axes[0].values = dateFormats
-
-  if (options.axes.length > 2) {
-    options.axes[2].side = 1
-  }
+  const options = createOptions({ plot, theme, width })
 
   function onCreate(chart: uPlot) {
```

Comment:
@szkiba is this function still required, I'm yet to see it being used.

Reply:
It is about to set second vertical axe to right side... if any